back/engines/free/frontend/spec/services/url_service_spec.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe Frontend::UrlService do
+  let(:service) { described_class.new }
+  let(:base_uri) { AppConfiguration.instance.base_frontend_uri }
+
+  describe '#model_to_url' do
+    let(:idea) { create(:idea) }
+    let(:internal_comment1) { create(:internal_comment, post: idea) }
+    let(:initiative) { create(:initiative) }
+    let(:internal_comment2) { create(:internal_comment, post: initiative) }
+    let(:user) { create(:user, locale: 'en') }
+
+    it 'returns the correct url for an internal comment on an idea' do
+      expect(service.model_to_url(internal_comment1, locale: user.locale))
```

Comment:
```suggestion
      expect(service.model_to_url(internal_comment1, locale: 'en'))
```
In this case, you don't need to create a user. Though not a big deal.